### PR TITLE
Add module/no-module support

### PR DESCRIPTION
**Pull request by sebastianbenz**
This PR adds the experimental module/nomodule runtime loading support.
ESM module loading support can be added via the `experimentEsm=true`
parameter.

### CODE REVIEW

**Comment by sebastianbenz** [comment]
FYI: @mdmower @patrickkettner

**Comment by sebastianbenz** [comment]
...and FYI @schlessera @westonruter - we should add this to the wp plugin once the validator supports mjs files.

**Comment by westonruter** [comment]
Filed issue for PHP Optimizer: https://github.com/ampproject/amp-wp/issues/4607

File: packages/optimizer/lib/transformers/RewriteAmpUrls.js [hunk 1]
```diff
@@ -101,6 +107,31 @@ class RewriteAmpUrls {
...
   }
 
+  _addEsm(scriptNode, preload) {
+    const esmScriptUrl = scriptNode.attribs.src.replace(/\.js$/, '.mjs');
+    if (preload) {
+      const preload = createElement('link', {
+        as: 'script',
+        crossorigin: 'anonymous',
+        href: esmScriptUrl,
+        rel: 'preload',
```
Comment (patrickkettner) [inline packages/optimizer/lib/transformers/RewriteAmpUrls.js:117]: have we considered using `modulepreload`?

Comment (kristoferbaxter) [reply packages/optimizer/lib/transformers/RewriteAmpUrls.js:117]: Its unsupported in Safari and Firefox :(

Comment (patrickkettner) [reply packages/optimizer/lib/transformers/RewriteAmpUrls.js:117]: is they are both included to supporting browsers try to prefetch twice?

**Comment by sebastianbenz** [comment]
Thanks for the review!

**Pull request merged**